test/unit/spec/PrimaryAuth_spec.js

```diff
@@ -2017,4 +2025,75 @@ function (_, $, Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, PrimaryAuthFo
       });
     });
   });
-});
+  
+  Expect.describe('Registration Flow', function () {
+    itp('does not show the registration button if settings.registration is not set', function () {
+      return setup().then(function (test) {
+        expect(test.form.registrationContainer().length).toBe(0);
+      });
+    });
+    itp('does not show the registration button if settings.registration is empty object', function () {
+      var registration =  {};
+      return setupRegistrationButton(registration).then(function (test) {
+        expect(test.form.registrationContainer().length).toBe(0);
+      });
+    });
+    itp('does not show the registration button if settings.registration properties are undefined', function () {
+      var registration =  undefined;
+      return setupRegistrationButton(registration).then(function (test) {
+        expect(test.form.registrationContainer().length).toBe(0);
+      });
+    });
+    itp('shows the registration button if settings.registration properties are configured', function () {
+      var registration =  {
+        click: function () {
+          window.location.href = 'http://www.test.com';
+        },
+        label: 'Dont have an account ?',
+        text: 'Register'
+      };
+      
+      spyOn(Okta, 'loc').and.callFake(function (key) {
```

Comment:
i18n changes https://github.com/okta/npm/pull/642 . I saw couple of other tests doing this to verify the text.